lib/arel_extensions/visitors/mssql.rb

```diff
@@ -292,7 +294,9 @@ def visit_ArelExtensions_Nodes_Format o, collector
               collector << 'DATEPART('
               collector << dir
               collector << LOADED_VISITOR::COMMA
+              collector << 'CONVERT(datetime' << LOADED_VISITOR::COMMA << ' ' if !o.time_zone.empty?
               collector = visit o.left, collector
+              collector << ") AT TIME ZONE 'UTC' AT TIME ZONE '#{o.time_zone}'" if !o.time_zone.empty?
```

Comment:
How safe is this?  I feel very uncomfortable, we should use a dedicated routine to output strings.  It looks like an invitation for SQL injection.

Reply:
Yikes. You're right. Sanitize the string before interpolation?

Reply:
You will be safe if you use `Arel::Nodes.build_quoted(o.time_zone)`.  Which you'll have to traverse.

Actually, this `build_quoted` could be done in the ctor.  So we no longer have to check for emptiness here, just for nil or not.